ts/client/database/databaseEngine.ts

```diff
@@ -0,0 +1,129 @@
+import { ZERO_BYTES32 } from "../../constants";
+import { TreeAtLevelIsFull } from "../../exceptions";
+import { Hashable, Vacant } from "../../interfaces";
+import { Hasher } from "../../tree";
+import { DBTree } from "../../tree/dbTree";
+import { Leaf, LeafFactoryFunc } from "../../tree/leaves/Leaf";
+import { StorageEngine, WithWitness } from "../storageEngine/interfaces";
+
+export interface Entry<Item> {
+    itemID: number;
+    item: Item;
+}
+
+export class DatabaseEngine<Item extends Hashable>
+    implements StorageEngine<Item> {
+    private tree: DBTree<Leaf<Item>>;
+    private items: { [key: number]: Item } = {};
+    private cache: { [key: number]: Item } = {};
+    private journal: Entry<Item>[] = [];
```

Comment:
Per comments from @ChihChengLiang , we don't really need journal or cache and its corresponding calls to flush such as `trueUpdate`. We can remove this and make the `commit` function a no-op and write directly to db.